bugbug/models/duplicate.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from imblearn.over_sampling import BorderlineSMOTE
+from sklearn.calibration import CalibratedClassifierCV
+from sklearn.compose import ColumnTransformer
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import LabelEncoder
+from sklearn.svm import LinearSVC
+
+from bugbug import bug_features, bugzilla, feature_cleanup
+from bugbug.model import BugCoupleModel
+
+
+class LinearSVCWithLabelEncoding(CalibratedClassifierCV):
+    def __init__(self, clf):
+        super().__init__(clf)
+        self._le = LabelEncoder()
+
+    def fit(self, X, y):
+        super().fit(X, y)
+        self._le.fit(y)
+
+
+class DuplicateModel(BugCoupleModel):
+    def __init__(self, lemmatization=False):
+        BugCoupleModel.__init__(self, lemmatization)
+
+        self.sampler = BorderlineSMOTE(random_state=0)
+        self.calculate_importance = False
+
+        cleanup_functions = [
+            feature_cleanup.responses(),
+            feature_cleanup.hex(),
+            feature_cleanup.dll(),
+            feature_cleanup.fileref(),
+            feature_cleanup.url(),
+            feature_cleanup.synonyms(),
+            feature_cleanup.crash(),
+        ]
+
+        self.extraction_pipeline = Pipeline(
+            [
+                ("bug_extractor", bug_features.BugExtractor([], cleanup_functions)),
+                (
+                    "union",
+                    ColumnTransformer([("text", self.text_vectorizer(), "text")]),
+                ),
+            ]
+        )
+
+        self.clf = LinearSVCWithLabelEncoding(LinearSVC())
+
+    def get_labels(self):
+        bugs = {}
+        count = 20000
```

Comment:
You can use https://docs.python.org/3/library/itertools.html#itertools.islice instead of implementing it yourself.
Something like `for bug_data in itertools.islice(bugzilla.get_bugs(), 0, 20000):` should work.